controllers/httproute_controller.go

```diff
@@ -254,6 +254,30 @@ func (r *HTTPRouteReconciler) reconcileHTTPRouteResource(ctx context.Context, ht
 		r.eventRecorder.Event(httpRoute.K8sObject(), corev1.EventTypeWarning, k8s.HTTPRouteventReasonFailedAddFinalizer, fmt.Sprintf("Failed add finalizer due to %v", err))
 	}
 
+	ok := r.areBackendRefsIpFamiliesValid(ctx, httpRoute.Inner())
+
+	if !ok {
+		httpRouteOld := httpRoute.Inner().DeepCopy()
+
+		setupRouteStatusParents(httpRoute.Inner())
+
+		updateRouteCondition(httpRoute, metav1.Condition{
+			Type:               string(gateway_api.RouteConditionAccepted),
+			Status:             metav1.ConditionFalse,
+			ObservedGeneration: httpRoute.Inner().Generation,
+			Reason:             string(gateway_api.RouteReasonUnsupportedValue),
+			Message:            fmt.Sprintf("Dual stack Service is not supported"),
+		})
+
+		if err := r.client.Status().Patch(ctx, httpRoute.Inner(), client.MergeFrom(httpRouteOld)); err != nil {
+			glog.Errorf("updateHTTPRouteStatus: Patch() received err %s", err)
+
+			return errors.Wrapf(err, "failed to update httproute status")
+		}
+
+		errors.New("Invalid IpFamilies, Lattice Target Group doesn't support dual stack ip addresses")
```

Comment:
This line will not do anything. You may want to return earlier because we can't really create target group.

Reply:
Thanks, forgot to put a `return`  there.

> I think creation of the service should trigger reconcile request anyways. Looks like this function should only determine if the ipFamily is valid.

I can create a `HTTPRoute` with `BackendRefs` to a non-existing Service right? In that case, the `Get` call would return an error which we want to ignore

Reply:
yeah I think it will eventually lead to lattice_retry on stack deployer later